### PR TITLE
Ensure MTX file ingest stores all expression values

### PR DESCRIPTION
**Pull request by knapii-developments**
This PR addresses the issue when parsing MTX files where genes that have more than one observed expression value are getting truncated and only storing the last value (or none at all). Test for Dense have been removed and will be addressed in [SCP-2567](https://broadworkbench.atlassian.net/browse/SCP-2567)

This PR satisfies [SCP-2055](https://broadworkbench.atlassian.net/browse/SCP-2055)

### CODE REVIEW

File: ingest/ingest_pipeline.py [hunk 1]
```diff
@@ -230,30 +226,39 @@ def load(
             return 1
         return 0
 
-    # @profile
-    def load_expression_file(self, models, is_gene_model=False):
-        collection_name = self.matrix.COLLECTION_NAME
-        # Creates operations to perform for bulk write
-        bulk_operations = list(map(lambda model: InsertOne(model), models))
+    def load_expression_file(self, gene_docs, data_array_documents):
+        self.error_logger.error(f'Starting to load expression file', extra=self.extra_log_params)
```
Comment (bistline) [inline ingest/ingest_pipeline.py:230]: Small nit - this should probably be a different log level other than `error`, but really we just want to ensure that it's logged somewhere.

File: ingest/mtx.py [hunk 2]
```diff
@@ -28,116 +33,134 @@
...
+            data_arrays.append(all_cell_model)
+        for mtx_gene_idx in range(int(self.mtx_description[0]) - 1):
+            exp_scores = []
+            exp_cells = []
+            id = ObjectId()
+            # Extract rows:str from mtx file that match current mtx_gene_idx
+            # matched_rows = [[N, K, M],[N, K, M],...N] where N = mtx_gene_idx
+            # Rows in mtx file are 1 based and represented as strings => have
+            # to add 1 to mtx_gene_idx and convert to string.
+            matched_rows = self.extract_mtx(
```
Comment (bistline) [inline ingest/mtx.py:119]: Very cool!

File: ingest/mtx.py [hunk 3]
```diff
@@ -28,116 +33,134 @@
...
         )
 
         mtx_ingest_file = IngestFiles(mtx_path, self.ALLOWED_FILE_TYPES)
-        mtx_file, self.mtx_local_path = mtx_ingest_file.resolve_path(mtx_path)
+        self.mtx_file, self.mtx_local_path = mtx_ingest_file.resolve_path(
+            mtx_path)
+        # A list ['N', 'K', 'M'] that represents a gene-barcode matrix where N
+        # is the gene index, M is the barcode index, and K is the expresion score
+        # for the given gene index
+        self.mtx_description = linecache.getline(
```
Comment (eweitz) [inline ingest/mtx.py:63]: Nifty, [`linecache` looks fast](https://stackoverflow.com/a/19190196) -- thanks for putting it on my radar.

File: tests/test_ingest.py [hunk 4]
```diff
@@ -102,115 +102,117 @@ def setup_ingest(self, args, mock_storage_client, mock_storage_blob):
...
-        ]
-        ingest = self.setup_ingest(args)[0]
+        return ingest, arguments, status, status_cell_metadata
 
-        models = ingest.load_args[0]
-        for model in models:
-            # Ensure that 'ObjectID' in model is removed
-            del model['_id']
-            self.assertEqual(model, gene_models[model['name']])
+    # def test_ingest_dense_matrix(self):
```
[the remark this answers is not in the document]
Comment (knapii-developments) [reply tests/test_ingest.py:109]: In [SCP-2567](https://broadworkbench.atlassian.net/browse/SCP-2567)

**Comment by knapii-developments** [comment]
> Good stuff! Cool use of `zgrep` and `linecache`.
> 
> Nothing blocking, but please note my concern about disabling dense matrix tests, and use of `error_logger` for non-errors. Beyond that, I made a handful of suggestions for easier readability.

**Pull request merged**